src/dataflow-types/src/types.rs

```diff
@@ -598,14 +603,43 @@ pub struct ProtobufEncoding {
     pub message_name: String,
 }
 
-/// Encoding in CSV format, with `n_cols` columns per row, with an optional header.
+/// Encoding in CSV format
 #[derive(Clone, Debug, Serialize, Deserialize)]
 pub struct CsvEncoding {
-    pub header_row: bool,
-    pub n_cols: usize,
+    pub columns: ColumnSpec,
     pub delimiter: u8,
 }
 
+impl CsvEncoding {
+    pub fn has_header(&self) -> bool {
+        matches!(self.columns, ColumnSpec::Header { .. })
+    }
+}
+
+/// What we know about the CSV columns
```

Comment:
Can we document this a bit better too? In particular we should probably call out that it changes the behavior of the decoder (i.e., that it determines whether the first row is a header row)